### PR TITLE
feat(forceContentUpdate): add forceContentUpdate

### PR DESCRIPTION
**Pull request by nzagorchev**

### CODE REVIEW

File: src/Leanplum.js [hunk 1]
```diff
@@ -136,6 +136,28 @@ export default class Leanplum {
...
+        response: function (response) {
+          let getVarsResponse = LeanplumRequest.getLastResponse(response);
+          if (LeanplumRequest.isResponseSuccess(getVarsResponse)) {
+            let values = getVarsResponse[Constants.KEYS.VARS];
+            let variants = getVarsResponse[Constants.KEYS.VARIANTS];
+            let actionMetadata = getVarsResponse[Constants.KEYS.ACTION_METADATA];
+            VarCache.applyDiffs(
+              values,
+              variants,
+              actionMetadata)
```
Comment (alexisoyama) [inline src/Leanplum.js]: Minor but this can be one line.

File: test/specs/testLeanplum.js [hunk 2]
```diff
@@ -406,5 +407,91 @@ Object.keys(testModes).forEach((mode) => {
...
+        delete require.cache[require.resolve(LEANPLUM_PATH)]
+      })
+
+      it('test setVariable forceContentUpdate', (done) => {
+        interceptRequest((request) => {
+          if (getAction(request) == 'getVars'){
+            request.respond(200, {
+              'Content-Type': 'application/json'
+            }, JSON.stringify(forceContentUpdateResponse))}
+            else {
```
Comment (alexisoyama) [inline test/specs/testLeanplum.js]: Format

**Comment by nzagorchev** [comment]
Should the response callback be as the startResponse one - returning success parameter, or the callback should be a variablesChanged callback without parameters as the ones in the iOS and Android SDKs?

**Comment by alexisoyama** [comment]
Great question. I personally think it should've been the start response one.. Need to verify with others for this one.

**Comment by nzagorchev** [comment]
Added the forceContentUpdate callback. It executes with a success parameter and mimics the callback in other SDKs.

File: src/Leanplum.js [hunk 3]
```diff
@@ -136,6 +136,30 @@ export default class Leanplum {
...
 
+  static forceContentUpdate(callback) {
+    LeanplumRequest.request(Constants.METHODS.GET_VARS,
+      new ArgsBuilder()
+      .add(Constants.PARAMS.INCLUDE_DEFAULTS, false), {
+        queued: false,
+        sendNow: true,
+        response: function (response) {
+          let getVarsResponse = LeanplumRequest.getLastResponse(response);
+          let success = false;
```
Comment (alexisoyama) [inline src/Leanplum.js]: We can use isResponseSuccess() here. That way we don't have to explicitly set it 'true' later.
Also it is easier to read ;).

File: src/Leanplum.js [hunk 4]
```diff
@@ -136,6 +136,30 @@ export default class Leanplum {
...
+          if (LeanplumRequest.isResponseSuccess(getVarsResponse)) {
+            let values = getVarsResponse[Constants.KEYS.VARS];
+            let variants = getVarsResponse[Constants.KEYS.VARIANTS];
+            let actionMetadata = getVarsResponse[Constants.KEYS.ACTION_METADATA];
+            VarCache.applyDiffs(values, variants, actionMetadata)
```
Comment (alexisoyama) [inline src/Leanplum.js]: Let's use the same format as the start's response.
VarCache.applyDiffs(
               startResponse[Constants.KEYS.VARS],
               startResponse[Constants.KEYS.VARIANTS],
               startResponse[Constants.KEYS.ACTION_METADATA])

**Comment by alexisoyama** [comment]
test this please

**Comment by alexisoyama** [comment]
test this please

**Comment by nzagorchev** [comment]
I squashed but I do not have permissions to merge. Please accept and merge.

**Comment by alexisoyama** [comment]
Done. Thanks Nikola!

**Pull request merged**